mne/io/snirf/_snirf.py

```diff
@@ -355,15 +421,48 @@ def natural_keys(text):
                 info["chs"][idx]["loc"][0:3] = midpoint
                 info["chs"][idx]["coord_frame"] = coord_frame
 
-                if (snirf_data_type in [1]) or (
-                    (snirf_data_type == 99999) and (ch_types[idx] == "fnirs_od")
+                # get data type specific info:
+                wve_idx = int(
+                    _correct_shape(
+                        np.array(dat.get(f"{ch_root}/wavelengthIndex", [1]))
+                    )[0]
+                )
+                if snirf_data_type == SNIRF_CW_AMPLITUDE or (
+                    snirf_data_type == SNIRF_PROCESSED and ch_types[idx] == "fnirs_od"
+                ):
+                    info["chs"][idx]["loc"][9] = fnirs_wavelengths[wve_idx - 1]
+                elif snirf_data_type in (
+                    SNIRF_TD_GATED_AMPLITUDE,
+                    SNIRF_TD_MOMENTS_AMPLITUDE,
                 ):
-                    wve_idx = int(
-                        _correct_shape(
-                            np.array(dat.get("nirs/data1/" + chan + "/wavelengthIndex"))
-                        )[0]
-                    )
                     info["chs"][idx]["loc"][9] = fnirs_wavelengths[wve_idx - 1]
+                    if snirf_data_type == SNIRF_TD_GATED_AMPLITUDE:
+                        bin_idx = int(
+                            _correct_shape(
+                                np.array(dat.get(f"{ch_root}/dataTypeIndex"))
+                            )[0]
+                        )
+                        info["chs"][idx]["loc"][10] = (
+                            fnirs_time_delays[bin_idx - 1]
+                            * fnirs_time_delay_widths[bin_idx - 1]
```

Comment:
I modified this to index into `fnirs_time_delay_widths` when it didn't before. It didn't error in either case, and it seems like one of them should have! So we still need to add a suitable SNIRF test file to mne-testing-data @JohnGriffiths (or others) if you have a tiny one you can share

Reply:
@julien-dubois-k this one is also for you :point_up:

Reply:
test file shared [here](https://github.com/mne-tools/mne-python/pull/11064#discussion_r1838886429), hopefully it helps with this case